structs.go

```diff
@@ -298,13 +306,106 @@ type Channel struct {
 
 	// ApplicationID of the DM creator Zeroed if guild channel or not a bot user
 	ApplicationID string `json:"application_id"`
+
+	// Voice region ID for the voice channel, automatic when set to null
+	RTCRegion *string `json:"rtc_region"`
+
+	// The camera video quality mode of the voice channel, 1 when not present
+	VideoQualityMode VideoQualityMode `json:"video_quality_mode,omitempty"`
+
+	// An approximate count of messages in a thread, stops counting at 50
+	MessageCount int `json:"message_count,omitempty"`
+
+	// An approximate count of users in a thread, stops counting at 50
+	MemberCount int `json:"member_count,omitempty"`
+
+	// Thread-specific fields not needed by other channels
+	ThreadMetadata *ThreadMetadata `json:"thread_metadata"`
+
+	// Thread member object for the current user, if they have joined the thread
+	// Only included on certain API endpoints
+	Member *ThreadMember `json:"member"`
+
+	// Default duration for newly created threads, in minutes,
+	// to automatically archive the thread after recent activity,
+	DefaultAutoArchiveDuration ArchiveDuration `json:"default_auto_archive_duration,omitempty"`
 }
 
 // Mention returns a string which mentions the channel
 func (c *Channel) Mention() string {
 	return fmt.Sprintf("<#%s>", c.ID)
 }
 
+// IsThread returns wether the specific channel is a thread
+func (c *Channel) IsThread() bool {
+	return c.Type == ChannelTypeGuildNewsThread || c.Type == ChannelTypeGuildPrivateThread || c.Type == ChannelTypeGuildPublicThread
+}
+
+// ThreadMetadata contains a number of thread-specific
+// channel fields that are not needed by other channel types.
+type ThreadMetadata struct {
+	Archived            bool            `json:"archived"`
+	AutoArchiveDuration ArchiveDuration `json:"auto_archive_duration"`
+	ArchiveTimestamp    Timestamp       `json:"archive_timestamp"`
+	Locked              bool            `json:"locked,omitempty"`
+}
+
+// ThreadMember is used to indicate whether a user has joined
+// a thread or not.
+type ThreadMember struct {
+	ID            string    `json:"id,omitempty"`
+	UserID        string    `json:"user_id,omitempty"`
+	JoinTimeStamp Timestamp `json:"join_timestamp"`
+	Flags         int       `json:"flags"`
+}
+
+// ThreadCreateData is the data used to create threads
+type ThreadCreateData struct {
+	// 2-100 character channel name
+	Name string `json:"name"`
+
+	// Duration in minutes to automatically archive the thread
+	// after recent activity.
+	AutoArchiveDuration ArchiveDuration `json:"auto_archive_duration"`
+
+	// Defaults to `PRIVATE_THREAD` in order to match the behavior
+	// when thread documentation was first published.
```

Comment:
I'm not sure that this comment is really relevant here given that we don't specify `omitempty`; there's no way for the "default" behavior on the API side to take effect. Perhaps having `omitempty` might make sense?